### PR TITLE
feat: add sticky segmented control in assets screen

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

This PR:
- add a new segmented control component (the existing one used on the QR screen is tightly coupled to the react navigation tabs implementation, i couldn't figure out how to easily extend it. it also looks different...)
- add animations for a sticky segmented control, where part of the list header is not sticky
- on tap of the segmented control, update a state value. displaying the positions isn't included in this PR.
- no unit tests yet because the segmented control doesn't do anything

Note: i need to change the Flatlist to a Sectionlist for the positions display, but Sectionlist doesn't allow a sticky header (only the section headers can be sticky). That is why we are not using a ListHeaderComponent here anymore.

### Test plan

iOS:

https://github.com/valora-inc/wallet/assets/20150449/84c2d455-bd5f-471f-9666-86138fa7667c

Android - the screen contrast isn't good here but there is a slight shadow when the header becomes sticky:

https://github.com/valora-inc/wallet/assets/20150449/3928cc89-b96d-49e9-8cf9-11e8d7ff3fb9

Android, without the segmented control:

https://github.com/valora-inc/wallet/assets/20150449/823113f7-ac43-49bd-817d-2c4d8b5333e8

### Related issues

- Fixes RET-734

### Backwards compatibility

Y

### CODE REVIEW

**Review by dievazqu — APPROVED** [review]
Looks so good!! Good job!

File: src/tokens/TokenBalances.tsx [hunk 1]
```diff
@@ -54,25 +63,68 @@ function TokenBalancesScreen({ navigation }: Props) {
...
+    },
+  })
+
+  const animatedHeaderOpacity = useDerivedValue(() => {
+    if (nonStickyHeaderHeight === 0) {
+      // initial render
+      return 0
+    }
+
+    const startAnimationPosition = nonStickyHeaderHeight - 44
```
[the remark this answers is not in the document]
Comment (kathaypacific) [reply src/tokens/TokenBalances.tsx:91]: yes i think so - dynamically measuring the height of the non sticky header ensures that we take into account the device font scaling / localisation. the hard coded values here are just the range of pixels over which the animation is happening, so the transition always starts / ends when the user has scrolled to within is 44px / 20px of the end of the header :)

**Review by jeanregisser — COMMENTED** [review]
Awesome! 🤩 

> on tap of the segmented control, update a state value. displaying the positions isn't included in this PR.
> no unit tests yet because the segmented control doesn't do anything

What do you think of adding a test for `SegmentedControl` itself that ensures tapping one of the provided values fires the `onChange` callback?

File: src/tokens/SegmentedControl.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+interface Props {
+  values: string[]
+  selectedIndex: number
+  onChange: (value: string, selectedIndex: number) => void
+}
+
+function SegmentedControl({ values, selectedIndex, onChange }: Props) {
+  const handleSelectOption = (value: string, index: number) => () => {
+    onChange(value, index)
+  }
```
[the remark this answers is not in the document]
Comment (kathaypacific) [reply src/tokens/SegmentedControl.tsx]: good idea, added

**Comment by kathaypacific** [comment]
> Awesome! 🤩
> 
> > on tap of the segmented control, update a state value. displaying the positions isn't included in this PR.
> > no unit tests yet because the segmented control doesn't do anything
> 
> What do you think of adding a test for `SegmentedControl` itself that ensures tapping one of the provided values fires the `onChange` callback?

i will add tests for switching the views in the token balances screen, i thought the segmented control would be included in the testing there. but it doesn't hurt to add a simple test here so i did

**Review by jeanregisser — APPROVED** [review]
🙌

**Pull request merged**